roles/hosted_engine_setup/tasks/bootstrap_local_vm/04_engine_final_tasks.yml

```diff
@@ -48,6 +48,18 @@
     with_items:
       - cloud-init-local
       - cloud-init
+  - name: Check if keycloak is configured
+    command: otopi-config-query query -k OVESETUP_KEYCLOAK_CORE/enable -f /etc/ovirt-engine-setup.conf
+    register: keycloak_configured_out
+    ignore_errors: true
+  - name: Set admin username
+    set_fact:
+      he_admin_username: >-
+        {{ 'admin@ovirt@internalsso'
+            if keycloak_configured_out.rc == 0 and keycloak_configured_out.stdout_lines[0] == 'True'
```

Comment:
AFAICT, we basically update otopi based on the ansible role and override the ansible role variables based on otopi.
Maybe update `he_admin_username` if `he_enable_keycloak` is true, and in otopi, update `he_enable_keycloak` to be false/true based on customer needs.

Reply:
Well, doesn't it cover also the scenario when you pause HE installation, SSH to engine VM and disable keycloak? If so, then that's a good fail safe.

Reply:
true. otherwise it's more or less equivalent, so let's keep it like this when already tested...